src/extractor/edge_based_graph_factory.cpp

```diff
@@ -368,181 +377,279 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
         util::Percent progress(log, m_node_based_graph->GetNumberOfNodes());
         // going over all nodes (which form the center of an intersection), we compute all
         // possible turns along these intersections.
-        for (const auto node_at_center_of_intersection :
-             util::irange(0u, m_node_based_graph->GetNumberOfNodes()))
+        std::mutex merge_mutex; // To control merging of all the data into the main data structures
+        std::uint64_t nodes_completed = 0;
+
+        const NodeID node_count = m_node_based_graph->GetNumberOfNodes();
+        NodeID current_node = 0;
+
+        tbb::filter_t<void, NodeID> intersection_ids(tbb::filter::serial_in_order,
```

Comment:
Just a quick idea here: Maybe it's faster to work on `NodeID` chunks? TBB has the `tbb::blocked_range<NodeID,  BLOCKSIZE>` abstraction for that. Could reduce the overhead a little bit, but I'm not sure.